### PR TITLE
Added argument parsing via boost program_options

### PR DESCRIPTION
**Pull request by CJRMoore**
Same argument parsing capabilities as AutoBub.

### CODE REVIEW

**Review by rethmecc — APPROVED** [review]

**Review by smfallows — APPROVED** [review]
Looks good. Do we want to continue setting these up for two approvals? Not that I mind at all, but we'll have to balance with the fact that waiting for all approvals could slow things down unnecessarily when more is happening in the future. At least one is nice, but beyond that it might just be up to whether the author thinks a specific person should take a look.

**Pull request merged**
